tracer/src/Datadog.Trace/AppSec/Transports/Http/HttpTransport.Core.cs

```diff
@@ -70,6 +69,41 @@ public void OnCompleted(Action completedCallback)
                 return Task.CompletedTask;
             });
         }
+
+        private readonly struct HeadersCollectionAdapter : IHeadersCollection
```

Comment:
I think I saw that code already in our codebase. It may not make sense, but should we refactor to avoid duplication? Or you really need a different implementation?

Reply:
There are currently 9 implantations of `IHeadersCollection` (including this one) and two others appear to be exactly the same as this one. I guess these were implemented as private structures because they don't fully implement the interface, so who ever created them didn't want them to be generally available, so I thought I'd carry on this tradition 😉  

Anyway, as it's not actually that difficult to implement all members, I did that and made the struct internal in 2403b6fd